rust/connlib/tunnel/src/gateway.rs

```diff
@@ -336,9 +354,10 @@ impl GatewayState {
 
         match self.next_expiry_resources_check {
             Some(next_expiry_resources_check) if now >= next_expiry_resources_check => {
-                self.peers
-                    .iter_mut()
-                    .for_each(|p| p.expire_resources(utc_now));
+                self.peers.iter_mut().for_each(|p| {
+                    p.expire_resources(utc_now);
+                    p.expire_nat(now)
+                });
```

Comment:
Eventually, it would be good to continue the model of `handle_timeout` all the way to the `ClientOnGateway` struct.